packages/mendel-atomic-css/index.js

```diff
@@ -0,0 +1,74 @@
+const vm = require('vm');
+const path = require('path');
+const m = require('module');
+const nodeModuleCache = {};
+
+function isNodeModule(filePath) {
+    return filePath.indexOf('node_modules') >= 0;
+}
+
+function runEntryInVM(entryProxy, sandbox, resolver) {
+    const filename = entryProxy.id;
+    const cacheable = isNodeModule(filename);
+    if (cacheable && nodeModuleCache[filename]) {
+        return nodeModuleCache[filename];
+    }
+
+    function localRequire(requireLiteral) {
+        // TODO remove hard coded "main"
+        const depNormId = entryProxy.deps[requireLiteral].main;
+        return runEntryInVM(resolver(depNormId), sandbox, resolver);
+    }
+
+    const exports = {};
+    const module = {exports};
+
+    // the filename is only necessary for uncaught exception reports to point to the right file
+    const nodeSource = vm.runInContext(m.wrap(entryProxy.source), sandbox, {filename});
+    // function (exports, require, module, __filename, __dirname)
+    nodeSource(exports, localRequire, module, filename, path.dirname(filename));
+
+    if (cacheable) {
+        nodeModuleCache[filename] = module.exports;
+    }
+
+    return module.exports;
+}
+
+function evalEntries(entries, mainEntry) {
+    const sandbox = {console};
+    vm.createContext(sandbox);
+
+    const entryMap = new Map();
+    entries.forEach(entry => entryMap.set(entry.normalizedId, entry));
+
+    return runEntryInVM(mainEntry, sandbox, (id) => entryMap.get(id));
+}
+
+module.exports = {
+    kind: 'gst',
+    predicate: function(entry) {
+        return !isNodeModule(entry.id) && entry.source.indexOf('getAtomicClasses') >= 0;
```

Comment:
Right now `isNodeModule` is OK, but should mendel itself only run this plugin against types that were assigned to this plugin on the mendelrc? Mono-repos and sub projects would benefit of the "node_module" restriction being removed.